text/src/autogluon/text/automm/data/process_text.py

```diff
@@ -35,6 +41,59 @@
 }
 
 
+def construct_augmenter(
+        augment_types: List[str],
+) -> naf.Sometimes:
+    """
+    Build up a text augmentor from the provided list of augmentation types
+    
+    Parameters
+    ----------
+    augment_types
+        A list of text augment types.
+
+    Returns
+    -------
+    A nlpaug sequantial flow.
+    
+    """
+    if(augment_types is None or len(augment_types) == 0):
+        return None
+
+    auglist = []
+    try:
+        nltk.data.find('tagger/averaged_perceptron_tagger')
+    except LookupError:
+        nltk.download('averaged_perceptron_tagger')
+    for aug_type in augment_types:
+        if '(' in aug_type:
+            trans_mode = aug_type[0:aug_type.find('(')]
+            kwargs = ast.literal_eval(aug_type[aug_type.find('('):])
+        else:
+            trans_mode = aug_type
+            kwargs = {}
+        if trans_mode == "synonym_replacement": 
+            kwargs['aug_src'] = 'wordnet'
+            try:
+                nltk.data.find('corpora/wordnet')
+            except LookupError:
+                nltk.download('wordnet')
+            try:
+                nltk.data.find('corpora/omw-1.4')
+            except LookupError:
+                nltk.download('omw-1.4')
```

Comment:
We may consider to later move these download to a separate `download_nltk_resources` function.